### DISCUSSION TITLE
How to you start your teaching demos?

### DISCUSSION THREAD

**Discussion by brownsarahm** [carpentries/trainers#95] [Show and tell]
We each have little tips & scripts for how we start demos. This is a place to share reminders you give trainees and practices you use for preparing yourself for demos as well.  

The [trainer guide](https://docs.carpentries.org/topic_folders/instructor_training/trainers_guide.html#running-a-teaching-demonstration) is the current official recommendations, but this is a place to discuss all the ways we all adapt and customize these guildelines.

**Comment by maneesha** [comment]
I have a script I've made for myself.  I think I shared a version of this at some point, but here it is again:

> **Timing:** 
> * :00 to :05 - introductions
> * :05 to :55 - 10 minutes per trainee.  1-2 minutes to get set up, 5 minute demo, 3-4 minutes feedback
> * :55 to 1:00 - questions/wrap-up
> 
> Welcome everyone 
> My name is Maneesha Sane and I work with The Carpentries.  I will be your demo session host for today. These demo sessions are run by our instructor trainers, some who are doing this as volunteers from their own sites, some like me who are on staff with The Carpentries. I always like when I get a chance to lead these teaching demo sessions because I get to see some of our up and coming instructors and start to get to know you all. Looks like we have a good group with us today, and I see you've chosen lessons from our various curricula. I'm also looking forward to seeing this range of topics for the demos.
> 
> Say something here about the group.  Examples:
> - I program mainly in Python, and I'm always excited to see how other people teach Python. 
> - OpenRefine is also something I want to be better at using, so I'm looking forward to seeing that lesson demo as well. 
> - I've only taught R a couple of times, so I know I'll learn from watching you.  
> - We have a great geographic representation - it's exciting to see The Carpentries' global reach!
> 
> So, on the etherpad, each of you has signed up and I am going to give you a specific section for you to teach from. You'll have a moment to get yourself oriented once I tell you where to teach from. I am going to give you 5 minutes to teach, I'll announce when time is up.  Don't worry if you're in the middle of a sentence -- remember, you're teaching five minutes of a lesson, not a five minute lesson.  Then we'll take a few minutes giving each other feedback on the etherpad, on the +/- and content/presentation scale that you used during your instructor training.  As the presenter you'll also have a chance to share any comments about what the experience was like.  
> 
> I want to acknowledge this is kind of an artificial environment.  In real life you'll never teach for just 5 minutes out of context.  You can assume that people  have been exposed to but haven't necessarily mastered  previous parts of the lessons.  So what I'll be looking for is generally that you are prepared, knowledgeable about what you'll teach, speak well, that you incorporate the pedagogical practices that you learned in instructor training, things like that.  
> 
> Do any of you have any questions?  
> 
> So with that, we'll just go down in the order we have people on the etherpad. [NAME], please take a minute to get yourself ready, and while you're doing that, I'll set my timer.
> 
> _Go through presentations and feedback from each trainee._
> 
> _Last 5 minutes for summary feedback and questions._
> 
> Thanks everyone for participating in the teaching demo today.  I'll email you back by tomorrow to let you know if you passed, or inviting you back to try again. If we do invite you back to try again, I'll give you some specific feedback on how to improve for next time. If you have any other questions along the way, my email is on the etherpad so please feel free to reach out.  Thanks again!

**Comment by jcoliver** [comment]
I have a template e-mail that I send to everyone a few days before the demo that includes the reminder about teaching _any_ episode. I add a little snippet for the particular lesson, reminding them about installing dependencies (i.e. R packages) and downloading data. My process is ready for automation, but I'm still copy-pasting. :smile:

**Reply by sstevens2** [reply]
I could also use some automation. I copy paste my short template (actually usually from the last email I sent) and update it.  I send it to all participants bcc so that the other participants don't have a record of each other's names and emails in their email.

I don't include the particular lessons they listed but do remind them that they should prep the whole lesson and I will pick the episode.  Anyone who has linked a specific episode in the etherpad, I reply to them directly and mention that they included a link to an episode and need to be prepped for the whole lesson.

**Comment by Denubis** [comment]
Here's my template email for teaching demos (somewhat more elaborated than the default)

```Hi <names>

According to https://pad.carpentries.org/teaching-demos you have signed up to give an online teaching demo at https://www.timeanddate.com/worldclock/fixedtime.html?msg=Carpentries+Teaching+Demo&iso=20210412T0000. I look forward to seeing you all there. Please double check your local time against the link -- UTC is very unintuitive. 

As part of this prep, I wanted to make it very clear that I may give you any segment of the lesson you prepared to teach, so you must be ready to teach any part of your chosen lesson. Some people prepare to teach 5 minutes from a particular section and nothing else, and these often have to reschedule as they seldom are assigned the section they have prepared for. A lesson corresponds to a single line in the lesson tables (https://software-carpentry.org/lessons/, http://www.datacarpentry.org/lessons/ , and https://librarycarpentry.org/lessons/) and a single repository on GitHub. Some lessons have supplementary modules, but you do not need to be prepared to teach the supplementary modules for your teaching demonstration.

 Make sure your environment is prepared and you're ready to teach, as if you were teaching in a workshop. I find it wise to print the lesson and highlight/annotate it as if I was preparing for a speech when I'm getting ready to run workshops. 

We will be using https://cuckoo.team/carpentries-teaching-demo as our timer.
The zoom call is: https://carpentries.zoom.us/my/carpentriesroom2
Our shared note-taking document and a copy of the rubric is at: https://docs.google.com/document/d/1_cmX6hwfoQrp3eSlTbWbTcHTOEGKxwFw7HRoRPVUEJs/edit?usp=sharing


I have the following lessons down:

name1 , urgent -- which LC lesson have you signed up for?
Name2, Library Carpentry, Open refine: https://librarycarpentry.org/lc-open-refine/
Name3, Software Carpentry, Python http://swcarpentry.github.io/python-novice-gapminder/
Name4, Software Carpentry, R, http://swcarpentry.github.io/r-novice-inflammation/
Name 5, Library Carpentry, Shell, https://librarycarpentry.org/lc-shell/

I am happy to answer any questions you wish to ask.

See you at the demo session,

-Brian
```

I also have my own template for shared notetaking: https://docs.google.com/document/d/1OR5vxyahdenh10xyv4bFI29NBRA4s6ufJdlT97vJvk4/edit that I feel prompts people well.

**Comment by sstevens2** [comment]
Added my pre-demo email info under Jeff's above.  Here is more detail about how I ~start~run the actual demo session.  

Before the demo, I pick out which episodes they will start out using the table for good places to start.  I add the episode link after the link they put in the etherpad.  I go ahead and paste all the episodes links in about 5 min before the start of the session.

At the beginning of the session, I tell everyone the plan for the session (intros then 5 min teaching demos and feedback), including the following announcements:
- I won't tell them if they pass or need to do another demo but will follow-up with an email,
- they should mute when they aren't talking (less of an issue now)
- on their turn they will get a couple min to setup and screenshare and I will use my phone to time 5 min to time them (and sometimes they won't hear it so I will say "bong" into the mic)
- I will ask them how they felt or feedback they have for themself when the 5 min is up and then ask for feedback from the room, and then give the feedback I have (+-PC rubric)

I then do a round of intros.  This actually takes a good amount if time but I think it helps everyone relax and feel more comfortable giving each other feedback.  Because this takes a bit of time, I can't do more than 5 teaching demos in an hour and sometimes have to cut down the verbal feedback between teaching demos if there are 5 demos because of time constraints.

Last thoughts on the rest of the session...
- I prompt but don't require the attendees to give each other verbal feedback and give verbal feedback myself between demos. If we run short on time, I ask them to give their feedback in chat and still give my feedback verbally.
- At the end if they all passed, I tell them they all passed and they won't get an email from me but I'll let the Carpentries know they finished this requirement for instructor checkout.  I also tell them if they've completed all the requirements they should get an email from the Carpentries telling them they are badged and if they don't and think they've completed the requirements to email checkout.
- I wrap up with telling them to keep my email address if they want to send me any questions and welcome them to the Carpentries community.

**Comment by jcoliver** [comment]
Here is the e-mail template that I use (note the "opportunity" to forget a copy/paste about specific lesson info):

Hi,

According to https://pad.carpentries.org/teaching-demos you have signed up to give an online teaching demo on Thursday 18 March 2021 19:00 UTC (See this link for your local time: https://www.timeanddate.com/worldclock/fixedtime.html?msg=Carpentries+Teaching+Demo&iso=20210318T19). I will be the Instructor Trainer running the session.

I wanted to make it clear that I may give you any segment of the lesson you prepared to teach, so you should be ready to teach any part of your chosen lesson. Some people prepare to teach 5 minutes from a particular section and nothing else, and they often have to reschedule as they seldom are assigned the section they have prepared for. A lesson corresponds to a single line in the lesson tables (https://software-carpentry.org/lessons/, http://www.datacarpentry.org/lessons/, and https://librarycarpentry.org/lessons/) and a single repository on GitHub. Some lessons have supplementary modules, but you do not need to be prepared to teach the supplementary modules for your teaching demonstration.

For example, if you have chosen The Unix Shell, I may assign you any episode listed at http://swcarpentry.github.io/shell-novice/.

**[insert lesson-specific snippet]**

Please visit the Etherpad (https://pad.carpentries.org/teaching-demos) a few minutes before the demo starts.

See you at the demo session,

**Comment by ragamouf** [comment]
[Teaching demo script](https://docs.google.com/document/d/1I33YtilyhtEh-fAR9Pr5PzaPWD06dFCfwstL2gf1L08/edit?usp=sharing)
